camayoc/tests/remote/rho/conftest.py

```diff
@@ -0,0 +1,149 @@
+        # now we will pass those parameters to another function
+        # to perform the scan and collect all testable facts.
+        # This will have the effect of the test being parametrized
+        # on the facts, not the profiles, so a test will run
+        # for each fact
+        metafunc.function.parametrize = [testable_facts(profiles_to_scan)]
+
+
+def testable_facts(profiles_to_scan):
+    """Perform scans on each profile and return facts.
+
+    This function will attempt to scan all profiles listed in config file. For
+    those that are successful, facts will be returned so the test may be
+    parametrized on them.
+
+    Additional data about the facts like what host, profile, auth, actual and
+    expected value are collected into a single dictionary with the scanned fact
+    and passed to the test that is marked as follows:
+
+        @pytest.mark.facts_needed
+        @pytest.mark.parametrize('fact', profiles())
+
+    """
+    profiles_to_scan = profiles_to_scan.args[1]
+    cfg = config.get_config()
+    facts_to_test = []
+    for profile in profiles_to_scan:
+        cwd = os.getcwd()
+        path = tempfile.mkdtemp()
+        for envvar in _XDG_ENV_VARS:
+            os.environ[envvar] = path
+        os.chdir(path)
+        reportfile = '{}-report.csv'.format(profile['name'])
+        try:
+            for auth in cfg['rho']['auths']:
+                if auth['name'] in profile['auths']:
+                    auth_add({
+                        'name': auth['name'],
+                        'username': auth['username'],
+                        'sshkeyfile': auth['sshkeyfile'],
+                    })
+
+            auths = ' '.join(item for item in profile['auths'])
+            hosts = ' '.join(item for item in profile['hosts'])
+            rho_profile_add = pexpect.spawn(
+                'rho profile add --name {} --auth {} --hosts {}'
+                .format(profile['name'], auths, hosts)
+            )
+            input_vault_password(rho_profile_add)
+            assert rho_profile_add.expect(
+                'Profile "{}" was added'.format(profile['name'])) == 0
+            assert rho_profile_add.expect(pexpect.EOF) == 0
+            rho_profile_add.close()
+            assert rho_profile_add.exitstatus == 0
+
+            rho_scan = pexpect.spawn(
+                'rho scan --profile {} --reportfile {} --facts all'
+                .format(profile['name'], reportfile),
+                timeout=300,
+            )
+            input_vault_password(rho_scan)
+            rho_scan.logfile = BytesIO()
+            assert rho_scan.expect(pexpect.EOF) == 0
+            logfile = rho_scan.logfile.getvalue().decode('utf-8')
+            rho_scan.logfile.close()
+            rho_scan.close()
+            assert rho_scan.exitstatus == 0, logfile
+            assert os.path.isfile(reportfile)
+        except (AssertionError, pexpect.exceptions.EOF):
+            # The scan failed for some reason.
```

Comment:
I think this is good for now but we may want to log the issues to help finding some errors later.